omnibus/config/software/python3.rb

```diff
@@ -66,24 +67,71 @@
-  source :url => "https://dd-agent-omnibus.s3.amazonaws.com/python-windows-#{version}-amd64.zip",
-         :sha256 => "045d20a659fe80041b6fd508b77f250b03330347d64f128b392b88e68897f5a0".downcase
+  else
+    dependency "vc_redist_14"
+
+    vcrt140_root = "#{Omnibus::Config.source_dir()}/vc_redist_140/expanded"
+    ###############################
+    # Setup openssl dependency... #
+    ###############################
+
+    # This is not necessarily the version we built, but the version
+    # the Python build system expects.
+    openssl_version = "3.0.15"
+    python_arch = "amd64"
+
+    mkdir "externals/openssl-bin-#{openssl_version}/#{python_arch}/include"
+    # Copy the import library to have them point at our own built versions, regardless of
+    # their names in usual python builds
+    copy "#{install_dir}/embedded3/lib/libcrypto.dll.a", "externals/openssl-bin-#{openssl_version}/#{python_arch}/libcrypto.lib"
+    copy "#{install_dir}/embedded3/lib/libssl.dll.a", "externals/openssl-bin-#{openssl_version}/#{python_arch}/libssl.lib"
+    copy "#{install_dir}/embedded3/lib/libssl.dll.a", "externals/openssl-bin-#{openssl_version}/#{python_arch}/libssl.lib"
+    # Copy the actual DLLs, be sure to keep the same name since that's what the IMPLIBs expect
+    copy "#{install_dir}/embedded3/bin/libssl-3-x64.dll", "externals/openssl-bin-#{openssl_version}/#{python_arch}/libssl-3.dll"
+    command "touch externals/openssl-bin-#{openssl_version}/#{python_arch}/libssl-3.pdb"
+    copy "#{install_dir}/embedded3/bin/libcrypto-3-x64.dll", "externals/openssl-bin-#{openssl_version}/#{python_arch}/libcrypto-3.dll"
+    command "touch externals/openssl-bin-#{openssl_version}/#{python_arch}/libcrypto-3.pdb"
+    # The applink "header"
+    copy "#{install_dir}/embedded3/include/openssl/applink.c", "externals/openssl-bin-#{openssl_version}/#{python_arch}/include/"
+    # And finally the headers:
+    copy "#{install_dir}/embedded3/include/openssl", "externals/openssl-bin-#{openssl_version}/#{python_arch}/include/"
+    # Now build python itself...
+
+    ###############################
+    # Build Python...             #
+    ###############################
+    # -e to enable external libraries. They won't be fetched if already
+    # present, but the modules will be built nonetheless.
+    command "PCbuild\\build.bat -e --pgo"
+    command "dir PCbuild/#{python_arch}/"
+    # Install the build artifact to their expected locations
+    copy "PCbuild/#{python_arch}/*.exe", "#{windows_safe_path(python_3_embedded)}/"
+    copy "PCbuild/#{python_arch}/*.dll", "#{windows_safe_path(python_3_embedded)}/"
+    mkdir "#{windows_safe_path(python_3_embedded)}/DLLs"
+    copy "PCbuild/#{python_arch}/*.pyd", "#{windows_safe_path(python_3_embedded)}/DLLs/"
+    mkdir "#{windows_safe_path(python_3_embedded)}/libs"
+    copy "PCbuild/#{python_arch}/*.lib", "#{windows_safe_path(python_3_embedded)}/libs"
+    copy "Lib", "#{windows_safe_path(python_3_embedded)}/"
```

Comment:
Where did this set of instructions come from? It seems to be missing some file and placing other in different locations compared to previous Agent versions.

For example
- `embedded3/sqlite3.dll` instead of `embedded3/DLLs/sqlite3.dll`.
- Missing `.cat` and `.ico` files, generally just non-pyd files,  in `DLLs`

Reply:
I mostly attempted to reproduce the previous file hierarchy, but indeed I didn't pay much attention to any non-openssl/python & non binary files.

I'm a bit wary about moving DLLs to another folder than the one it's been generated into since I don't know how python will attempt to load them. I can definitely attempt to put sqlite & other dependencies in `embedded3/DLLs/` though